app/models/framework_response.rb

```diff
@@ -112,4 +103,24 @@ def self.recursive_tree
   def set_responded_value
     self.responded = true
   end
+
+  def update_response_transaction(new_value)
+    retried ||= false
```

Comment:
I'm not convinced this will work with a local variable within this method - i.e. as soon as the method exits then `retried` falls out of scope and is eligible for garbage collection. Perhaps better to move this to an instance method in the class instead - the intent and behaviour should then be a bit clearer I think.

Reply:
so `retried` is memoized so it will remain in scope and the behaviour is as expected. However I've moved it to an instance method for clarity as suggested